corehq/ex-submodules/casexml/apps/phone/xml.py

```diff
@@ -121,10 +121,12 @@ def get_registration_element(restore_user):
     return root
 
 
-def get_registration_element_for_case(restore_user, case_id):
+def get_registration_element_for_case(restore_user, case_id, username):
+    if not username:
+        username = restore_user.username
     root = safe_element("Registration")
     root.attrib = {"xmlns": USER_REGISTRATION_XMLNS}
-    root.append(safe_element("username", restore_user.username))
+    root.append(safe_element("username", username))
```

Comment:
i don't know enough about this, but is this the right place to make the change to fix the bug? if we do this then username/password will be mismatched and could be misleading to someone picking up this code later.

Reply:
This is where the value that ends up in the form XML `<username>` block comes from and working around that would involve some pretty ugly hacking into the mobile engine that I'd prefer to avoid. Given that only Formplayer will ever use this route and Formplayer does not use the `password` value (and I don't think the mobile does either, though double checking this) I think we don't have to worry about this - and in fact I think we can remove `password` altogether here.

Reply:
ok, if we remove password then i think it's fine. but i don't think we should have a mismatched username and password in the same data structure